src/App.js

```diff
@@ -20,9 +24,34 @@ const [message, setMessage] = useState('');
 
   }, []);
 
+  const ProtectedRoute = ({ children }) => {
+    // Placeholder to actually check if user is authenticated
+    const useAuth = () => {
+      if (Math.random() > 0.5) {
+        return true;
+      }
+      return false;
+    }
+
+    const user = useAuth();
+    if (!user) {
+      // User is not authenticated
+      return <Navigate to="/login" />;
+    }
+    return children;
+  };
+
   return (
     <>
       <h1>{message}</h1>
+      <ProtectedRoute>
+        <Routes>
+          <Route exact path="/" element={<HomePage />} />
+        </Routes>  
+      </ProtectedRoute> 
```

Comment:
Rather than have ProtectedRoute wrap around the Routes component, let's have it wrap around the HomePage component. This should cut down on unnecessary rerendering.